src/Core/StaticConfigTrait.php

```diff
@@ -112,13 +112,31 @@ public static function setConfig($key, $config = null): void
      * Reads existing configuration.
      *
      * @param string $key The name of the configuration.
-     * @return mixed Configuration data at the named key or null if the key does not exist.
+     * @return mixed|null Configuration data at the named key or null if the key does not exist.
      */
     public static function getConfig(string $key)
     {
         return static::$_config[$key] ?? null;
     }
 
+    /**
+     * Reads existing configuration for a specific key.
+     *
+     * The config value for this key must exist, it can never be null.
+     *
+     * @param string $key The name of the configuration.
+     * @return mixed|null Configuration data at the named key.
+     * @throws \InvalidArgumentException If value does not exist.
+     */
+    public static function getConfigOrFail(string $key)
+    {
+        if (!isset(static::$_config[$key])) {
```

Comment:
Would be better to avoid the negation and instead return from the `if` block and move the exception below it.

Reply:
That would go abit against the flow of normal bail early principle.
The happy case goes without indentation to the last line of normal execution IMO.
But either way would be no big deal and is small semantics.

Reply:
The negation is an actual extra operation, the rest are just semantics.

Reply:
The operation has to be done, either way, the bool cast is nano opt IMO, and the code flow cleaner as per normal standards IMO. Bail early is what we should consistently use, having the happy case following afterwards as clear happy path.